base/number.jl

```diff
@@ -63,4 +63,6 @@ zero{T<:Number}(::Type{T}) = convert(T,0)
 one(x::Number)  = oftype(x,1)
 one{T<:Number}(::Type{T}) = convert(T,1)
 
+_default_type(::Type{Number}) = Int
```

Comment:
What is this change for?  Note that `_default_type(T::Union{Type{Real},Type{AbstractFloat}}) = Float64` in `float.jl` ... it seems a bit odd that `_default_type(Real) == Float64` but `_default_type(Number) == Int`.